### PR TITLE
Add broken test for record destructuring

### PR DESCRIPTION
**Pull request by cknitt**
Tests for #704.

The first two are working fine, the second two not.

### CODE REVIEW

File: analysis/tests/src/Destructuring.res [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+// let {} = x
+//      ^com
+
+let f = (x: x) => {
+  let {name, } = x
+  //         ^com
+  name
+}
+
+// let f = (x: x) => { let {} = x }
```
Comment (zth) [inline analysis/tests/src/Destructuring.res:17]: Probably better to put just the destructure inside of a comment and in the uncommented fn body above - a full function without compilation won't work even after we fix the underlying issues here.

Comment (cknitt) [reply analysis/tests/src/Destructuring.res]: Better now?

Comment (zth) [reply analysis/tests/src/Destructuring.res]: Yes, great.

**Comment by zth** [comment]
Fixed. My mistake was I only targeted structure items, whereas we now target all let bindings. Should make destructuring work in more cases. @cristianoc care to have a quick look?

**Review by cristianoc — APPROVED** [review]
Looking great, from what I can see.

**Pull request merged**